test/instances/aave/AaveL2Hyperdrive.t.sol

```diff
@@ -0,0 +1,461 @@
+            );
+            assertApproxEqAbs(
+                convertToShares(AWETH.balanceOf(trader)),
+                traderBalancesBefore.sharesBalance +
+                    convertToShares(baseProceeds),
+                2
+            );
+        }
+    }
+
+    /// Getters ///
+
+    function test_getters() external view {
+        assertEq(
+            address(IAaveL2Hyperdrive(address(hyperdrive)).vault()),
+            address(POOL)
+        );
+        assertEq(
+            hyperdrive.totalShares(),
+            AWETH.balanceOf(address(hyperdrive))
+        );
+    }
+
+    /// Price Per Share ///
+
+    function test__pricePerVaultShare(uint256 sharesPaid) external {
+        // Ensure that the share price is the expected value.
+        uint256 totalSupply = AWETH.totalSupply();
+        uint256 scaledTotalSupply = AWETH.scaledTotalSupply();
+        uint256 vaultSharePrice = hyperdrive.getPoolInfo().vaultSharePrice;
+        assertEq(vaultSharePrice, totalSupply.divDown(scaledTotalSupply));
+
+        // Ensure that the share price accurately predicts the amount of shares
+        // that will be minted for depositing a given amount of shares. This will
+        // be an approximation.
+        sharesPaid = sharesPaid.normalizeToRange(
+            2 *
+                convertToShares(
+                    hyperdrive.getPoolConfig().minimumTransactionAmount
+                ),
+            convertToShares(HyperdriveUtils.calculateMaxLong(hyperdrive))
+        );
+        uint256 hyperdriveSharesBefore = convertToShares(
+            AWETH.balanceOf(address(hyperdrive))
+        );
+        openLong(bob, sharesPaid, false);
+        assertApproxEqAbs(
+            AWETH.balanceOf(address(hyperdrive)),
+            (hyperdriveSharesBefore + sharesPaid).mulDown(vaultSharePrice),
+            1e4
+        );
+    }
+
+    /// Helpers ///
+
+    function advanceTime(
+        uint256 timeDelta,
+        int256 variableRate
+    ) internal override {
+        // Get the normalized income prior to updating the time.
+        uint256 reserveNormalizedIncome = POOL.getReserveNormalizedIncome(
+            address(WETH)
+        );
+
+        // Advance the time.
+        vm.warp(block.timestamp + timeDelta);
+
+        // Accrue interest in the Aave pool. Since the vault share price is
+        // given by `getReserveNormalizedIncome()`, we can simulate the accrual
+        // of interest by multiplying the total pooled ether by the variable
+        // rate plus one.
+        uint256 normalizedTime = timeDelta.divDown(365 days);
+        reserveNormalizedIncome = variableRate >= 0
+            ? reserveNormalizedIncome +
+                reserveNormalizedIncome.mulDown(uint256(variableRate)).mulDown(
+                    normalizedTime
+                )
+            : reserveNormalizedIncome -
+                reserveNormalizedIncome.mulDown(uint256(-variableRate)).mulDown(
+                    normalizedTime
+                );
```

Comment:
We've been switching to using `HyperdriveUtils.calculateInterest` in the other yield source tests. Could you update this to use that function?